src/app/store/keys/key.actions.ts

```diff
@@ -40,10 +41,7 @@ function setWalletEncryptionPassword(args: {
     }
 
     async function doesStacksAddressHaveBnsName(address: string) {
-      const resp = await stxClient.namesApi.getNamesOwnedByAddress({
-        address,
-        blockchain: 'stacks',
-      });
+      const resp = await fetchNamesForAddress(stxClient, address, true);
```

Comment:
Why are we setting this to`isTestnet: true`?

Reply:
Interesting - I honestly don't recall. This is probably not the logic you'd want.

Looking at where `setWalletEncryptionPassword` is called, it appears that `stxClient` is for the current network. However, is that ideal? Should whether the account 'exists' be dependent on the current network, or just mainnet?

Either way, we can add a `isTestnet` flag to `setWalletEncryptionPassword` and fetch that state from the store. If you wanted to change the logic to only lookup names on mainnet, that would require a different change to get the mainnet client in `useKeyActions`, ie

```ts
const { isTestnet } = useCurrentNetworkState();

// ...

return dispatch(keyActions.setWalletEncryptionPassword({ password, stxClient, btcClient, isTestnet }));
```

Reply:
> Should whether the account 'exists' be dependent on the current network, or just mainnet?

As this is for restoring highest account index, I figured checking mainnet is more important. Checking testnet as well would double amount of requests and it's pretty heavy as is.